Ghidra/Features/Decompiler/src/decompile/cpp/pcodecompile.cc

```diff
@@ -622,18 +622,10 @@ vector<OpTpl *> *PcodeCompile::assignBitRange(VarnodeTpl *vn,uint4 bitoffset,uin
   uint4 smallsize = (numbits+7)/8; // Size of input (output of rhs)
   bool shiftneeded = (bitoffset != 0);
   bool zextneeded = true;
-  uintb mask = (uintb)2;
-  int4 masknumbits = sizeof(mask) * 8;
-  if (numbits - 1 < masknumbits)
-    mask <<= (numbits - 1);
-  else
-    mask = 0;
-  --mask;
-  if (bitoffset < masknumbits)
```

Comment:
Sorry, I should have tested more examples before suggesting, this impl is correct over the one I suggested, mine didn't handle the `--` correctly looks like, I would revert back to yours.  This does look like a weird mask value if this `if` fails though (all bits, but I think it'd be no bits, right?).

Reply:
Hmm, I see what you mean. I wonder whether that distinction in the original code is intentional though. As a caller, if I supplied `numbits` to be a really high number to this function call, I think it'd be weird for the mask to have all the bits set (`0xFFF...`) which is that the old code was doing.